linter/linter.go

```diff
@@ -278,13 +278,35 @@ func (l *Linter) factoryRootStatements(vcl *ast.VCL, ctx *context.Context) []ast
 			}
 			statements = append(statements, stmt)
 		case *ast.SubroutineDeclaration:
-			if err := ctx.AddSubroutine(t.Name.Value, &types.Subroutine{Decl: t, Body: t.Block}); err != nil {
-				e := &LintError{
-					Severity: ERROR,
-					Token:    t.Name.GetMeta().Token,
-					Message:  err.Error(),
+			if t.ReturnType != nil {
```

Comment:
Conversely, we need to lint the fastly builtin subroutine (e.g `vcl_recv`) should not have a return type so we should lint them.

```vcl
sub vcl_recv BOOL {  // should lint error
  ...
}
```

Or, it can raise errors in the parser phase.

Reply:
fixed in https://github.com/ysugimoto/falco/pull/57/commits/da98ba23e804ef06a32ef3367f3c35bc6b3f15e2